### DISCUSSION TITLE
输入vi指令   输入内容始终会少一行

### DISCUSSION THREAD

**Discussion by yueshangyueshang** [sourcelair/xterm.js#4872] [Q&A]
哈喽，如果可以帮我看看，输入vi a.txt的时候，少了一行，aaa不见了
  const initTerm = () => {
    const terminal = new Terminal({
      rendererType: 'canvas', //渲染类型
      rows: rows.value, //行数
      // cols: cols.value, // 设置之后会输入多行之后覆盖现象
      convertEol: false, //启用时，光标将设置为下一行的开头
      fontSize: 14, //字体大小
      disableStdin: false, //是否应禁用输入。
      cursorStyle: 'block', //光标样式
      // cursorBlink: true, //光标闪烁
      scrollback: 30, //终端中的回滚量
      tabStopWidth: 4,
      theme: {
        foreground: 'white', //字体
        background: '#060101', //背景色
        cursor: 'help' //设置光标
      }
    });
    const attachAddon = new AttachAddon(socket.value);
    const fitAddon = new FitAddon();
    terminal.loadAddon(attachAddon);
    terminal.loadAddon(fitAddon);
    terminal.open(terminalRef.value);
    terminal.focus();
    //限制和后端交互，只有输入回车键才显示结果
    terminal.prompt = () => {
      terminal.write('\r\n$ ');
    };
    terminal.prompt();

![微信图片_20231108111605](https://github.com/xtermjs/xterm.js/assets/129679905/de95719e-bb35-423e-9385-53f48f5517c5)

不知道从哪前端入手还是后端

**Comment by tisilent** [comment]
试试cat .显示什么.

**Reply by yueshangyueshang** [reply]
已解决，后端的问题，谢谢

**Reply by MrLeungE1** [reply]
你好，我有一个问题想问您，我在命令行中输入 vi text进入到Vi界面之后，按上下键会运行外部命令逻辑(切换命令)，请问您有好的解决方法吗？

**Reply by wings-of-gold** [reply]
> 已解决，后端的问题，谢谢

请问，怎么解决的呢？
